pkg/controller/image-stream-tag-reconciler/reconciler.go

```diff
@@ -226,6 +257,8 @@ func (r *reconciler) currentHEADForBranch(br *branchReference) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("fauld to git rev-parse %s: %w", br.branch, err)
 	}
+	// RevParse output contains a trailing newline
+	branchHEADRef = strings.TrimSpace(branchHEADRef)
 
 	return branchHEADRef, nil
 }
```

Comment:
i love `getPublishJob` ... lol

Reply:
Is that irony? Its somewhat stupid and this should be part of some api package but it seems to work. It creates some implicit assumptions which is a bit unfortunate, so if you can come up with a better approach I am totally open for that.

Reply:
No I think your approach is appropriate (but sad we do not have something better)